src/main/kotlin/sidechain/eth/util/Crypto.kt

```diff
@@ -1,29 +1,31 @@
 package sidechain.eth.util
 
-import config.EthereumConfig
-import config.EthereumPasswords
 import notary.endpoint.eth.AmountType
+import org.web3j.crypto.ECKeyPair
 import org.web3j.crypto.Hash
-import org.web3j.protocol.admin.Admin
-import org.web3j.protocol.http.HttpService
+import org.web3j.crypto.Sign
 import org.web3j.utils.Numeric
 import java.math.BigInteger
 
-
 /**
  * Signs user-provided data with predefined account deployed on local Parity node
  * @param toSign data to sign
  * @return signed data
  */
-fun signUserData(ethereumConfig: EthereumConfig, ethereumPasswords: EthereumPasswords, toSign: String): String {
-    // TODO luckychess 26.06.2018 D3-100 find a way to produce correct signatures locally
-    val deployHelper = DeployHelper(ethereumConfig, ethereumPasswords)
-
-    val admin = Admin.build(HttpService(ethereumConfig.url))
-    admin.personalUnlockAccount(deployHelper.credentials.address, ethereumPasswords.credentialsPassword).send()
-    // TODO: signalize about error if unlock fails
+fun signUserData(ecKeyPair: ECKeyPair, toSign: String): String {
 
-    return admin.ethSign(deployHelper.credentials.address, toSign).send().signature
+    // Message from hex to bytes
+    val dat = Numeric.hexStringToByteArray(toSign)
+    // Add ethereum signature format
+    val to_sign = ("\u0019Ethereum Signed Message:\n" + (dat.size)).toByteArray()
+    val data_sign = to_sign.plus(dat)
+    val signature = Sign.signMessage(data_sign, ecKeyPair)
+    // Combine in the signature
+    var res = Numeric.toHexString(signature.r)
+    res = res.plus(Numeric.toHexString(signature.s).substring(2))
+    //  The v is always either 27, or 28 - need to convert to 00 or 01
+    res = res.plus("0" + Integer.toString(signature.v.toInt() - 27))
```

Comment:
what is 27? seems like magic constant. can we create named constant for this?

Reply:
It is Ethereum magic constant, we gonna is only here, so maybe there is no need for naming